### PR TITLE
Final changes in matrix_multiplication

### PR DESCRIPTION
**Pull request by emiljanogj**
Fixes a part of #64

### CODE REVIEW

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: Indent the comments to the same level (remove space before the first comment)

**Review by afd — APPROVED** [review]
Approved with one nit.

I am running the bots now - if they pass, I will merge once you fix the nit.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: Nit: indent this the same as the rest.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: When reviewing your other PR I realised that this compute shader isn't quite right: every workgroup unblocks the whole matrix.

Can you revise so that each workgroup takes care of unblocking a particular submatrix, so that the unblockings are disjoint and happen in parallel?

**Review by afd — CHANGES_REQUESTED** [review]
Actually when reviewing your other PR I noticed an issue with the compute shader for matrix unblocking (plus another small nit) - can you look into that?

Thanks

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: I'm fine with the new way you have expressed this, if you like it better. But I was also OK with the way you had done it before, having each thread take care of a single element.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: Nit: space between 16x16 and array.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: As above: you could revert this if you want; I don't mind either way.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (afd) [inline examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: This looks correct.

An alternative is to use a 16x16 set of workgroups and to have each invocation take care of a single submatrix element.

I'll leave it to you to decide what's best.

**Review by afd — APPROVED** [review]
One nit, and a couple of other comments. I'll merge without further review once you fix the nit and decide whether or not you want to go with the 16x16 workgroups or 4x4 workgroups approach.

**Comment by afd** [comment]
@EGJ1996 I don't feel the need to do another round of review on this, but I would like you to fix the nits and resolve the other comments with an indication as to what you decided to do. Please let me know when that's done.

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (emiljanogj) [reply examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: @afd I changed the workgroup number to 16x16 and replaced the nested for loops with:

```array_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data[row][col];```

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (emiljanogj) [reply examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: Changed this back to 16 16 1

File: examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap
Comment (emiljanogj) [reply examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap]: Same, changed the workgroups to 16x16 and replaced the nested for-loop by:
```res[row][col] = blocked_res[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE];```

**Review by afd — APPROVED** [review]
Thank you for this contribution!

**Pull request merged**
